libecl/src/ecl_grid.c

```diff
@@ -1431,20 +1431,21 @@ static void ecl_grid_taint_cells( ecl_grid_type * ecl_grid ) {
 
 
 static void ecl_grid_free_cells( ecl_grid_type * grid ) {
+  if (grid->cells) {
```

Comment:
What about rather flipping this around, and just return early? I think this makes it easier to read.